src/pyroboplan/planning/utils.py

```diff
@@ -2,6 +2,82 @@
 
 import numpy as np
 
+from ..core.utils import (
+    check_collisions_at_state,
+    check_collisions_along_path,
+    configuration_distance,
+    get_random_state,
+)
+
+from itertools import product
+
+
+def extend_robot_state(q_parent, q_sample, max_connection_distance):
+    """
+    Determines an incremental robot configuration between the parent and sample states, if one exists.
+
+    Parameters
+    ----------
+        q_parent : array-like
+            The starting robot configuration.
+        q_sample : array-like
+            The candidate sample configuration to extend towards.
+        max_connection_distance : float
+            Maximum angular distance, in radians, for connecting nodes.
+
+    Returns
+    -------
+        array-like
+            The resulting robot configuration, or None if it is not feasible.
+    """
+    q_diff = q_sample - q_parent
+    q_increment = max_connection_distance * q_diff / np.linalg.norm(q_diff)
+
+    q_cur = q_parent
+    # Clip the distance between nearest and sampled nodes to max connection distance.
+    # If we have reached the sampled node, then we just check that.
+    if configuration_distance(q_cur, q_sample) > max_connection_distance:
+        q_extend = q_cur + q_increment
+    else:
+        q_extend = q_sample
+
+    # Then there are no collisions so the extension is valid
+    return q_extend
+
+
+def has_collision_free_path(q1, q2, max_angle_step, model, collision_model):
+    """
+    Determines an incremental robot configuration between the parent and sample states, if one exists.
```

Comment:
Is this comment valid? Or is this copy-pasta

Reply:
I never claimed to be a smart person.